src/pages/Home/Home.tsx

```diff
@@ -87,6 +88,10 @@ const Home = () => {
     );
   }, [refresh, filterData, shelters.filters, shelters.page, shelters.perPage]);
 
+  const handleRefreshClick = useCallback(() => {
+    setForceRefresh((prev) => !prev);
+  }, []);
+
```

Comment:
Por qual motivo precisou de `useCallback` aqui? Pelo que vejo, você só está fazendo atualização de estado e é apenas mudança de valor booleano, não tem praticamente custo algum aqui para precisar de memoize nessa função (ela nem tem comportamento que muda e acordo com alguma situação). Isso está muito parecendo overengineering querendo resolver um problema de performance onde não existe problema algum.
ref: https://react.dev/reference/react/useCallback#should-you-add-usecallback-everywhere